python_modules/dagster/dagster/_core/events/__init__.py

```diff
@@ -1519,6 +1543,20 @@ def build_asset_materialization_planned_event(
         )
         return event
 
+    @staticmethod
+    def build_asset_failed_to_materialize_event(
+        job_name: str,
+        step_key: str,
```

Comment:
step_key is deprecated, and i think i should be able to get the step_handle when these events are emitted. I see other fns in the class take a step_key and were written more recently than step_key was deprecated. So just want to double check what is the right thing to do here

Reply:
If we're going to emit these in the event log consumer daemon, I think you can have pipeline failure events that trigger the emitting of these failure events.  In that case, I'm pretty sure you will not have a step_key value to pass in here.

Reply:
I think the planned event always has a step key though, which is ultimately what we are using right? But maybe in the airlift case we will be emitting these when there was not a step key

Reply:
step_key being deprecated is news to me. seems like we would need to add step_handle to the planned event to be able to use step_handle here?

Reply:
> But maybe in the airlift case we will be emitting these when there was not a step key

good point. I can make`step_key` optional